openHab 2 CometVisu PHP binding not working

Hi,

even if loaded over the PaperUI the cometvisu PHP is not working. See KNX forum thread:

[KNX Userforum Thread](https://knx-user-forum.de/forum/supportforen/cometvisu/954821-probleme-mit-openhab2-beta-3-und-cometvisu-0-9-2)

We tried several things. But all php related scripts like editor, icon overview or rss-log are not working. Only an blank page is shown.

regards René
##

This is more an osgi problem I think. What I can say is that, when the php-bundle has been loaded the cometvisu bundle does not get notified. I added a debug message in the setPHProvider method that is defined in the osgi-XML file, but all I see is:

```
18:38:52.018 [DEBUG] [org.openhab.ui.cometvisu            ] - BundleEvent INSTALLED - org.openhab.ui.cometvisu
18:38:52.030 [DEBUG] [org.openhab.ui.cometvisu.php        ] - BundleEvent INSTALLED - org.openhab.ui.cometvisu.php
18:38:52.325 [DEBUG] [org.openhab.ui.cometvisu.php        ] - BundleEvent RESOLVED - org.openhab.ui.cometvisu.php
18:38:52.464 [DEBUG] [org.openhab.ui.cometvisu.php        ] - BundleEvent STARTING - org.openhab.ui.cometvisu.php
18:38:52.466 [DEBUG] [org.openhab.ui.cometvisu.php        ] - BundleEvent STARTED - org.openhab.ui.cometvisu.php
18:38:52.512 [DEBUG] [org.openhab.ui.cometvisu.php        ] - ServiceEvent REGISTERED - {org.openhab.ui.cometvisu.php.PHProvider}={component.name=org.openhab.ui.cometvisu.php, component.id=230, service.id=390, service.bundleid=220, service.scope=bundle} - org.openhab.ui.cometvisu.php
18:38:52.761 [ERROR] [core.karaf.internal.FeatureInstaller] - Failed uninstalling 'openhab-ui-cometvisu': Feature named 'openhab-ui-cometvisu/0.0.0' is not installed
```

The debug message is missing, I have no clue what the error means. Maybe some osgi expert can have a look at this.

--- a/addons/ui/org.openhab.ui.cometvisu/OSGI-INF/cv-frontend-app.xml
+++ b/addons/ui/org.openhab.ui.cometvisu/OSGI-INF/cv-frontend-app.xml
@@ -9,7 +9,7 @@
     http://www.eclipse.org/legal/epl-v10.html
 
 -->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" configuration-policy="optional" immediate="true" name="org.openhab.cometvisu">
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.2.0" configuration-policy="optional" immediate="true" name="org.openhab.cometvisu">
    <implementation class="org.openhab.ui.cometvisu.servlet.CometVisuApp"/>
    <service>
       <provide interface="org.openhab.ui.cometvisu.servlet.CometVisuApp"/>
@@ -21,6 +21,6 @@
    <reference bind="addIconProvider" cardinality="1..n" interface="org.eclipse.smarthome.ui.icon.IconProvider" name="IconProvider" policy="dynamic" unbind="removeIconProvider"/>
    <reference bind="addPersistenceService" cardinality="0..n" interface="org.eclipse.smarthome.core.persistence.PersistenceService" name="PersistenceService" policy="dynamic" unbind="removePersistenceService"/>
    <reference bind="setEventPublisher" cardinality="1..1" interface="org.eclipse.smarthome.core.events.EventPublisher" name="EventPublisher" policy="static" unbind="unsetEventPublisher"/>
-   <reference bind="setPHProvider" cardinality="0..1" interface="org.openhab.ui.cometvisu.php.PHProvider" name="PHProvider" policy="static" unbind="unsetPHProvider"/>
+   <reference bind="setPHProvider" cardinality="0..1" interface="org.openhab.ui.cometvisu.php.PHProvider" name="PHProvider" policy="dynamic" policyOption="greedy" unbind="unsetPHProvider"/>
    <property name="service.pid" type="String" value="org.openhab.cometvisu"/>
 </scr:component>

